build-system/compile/compile.js

```diff
@@ -316,7 +316,11 @@ function generateCompilerOptions(outputDir, outputFilename, options) {
     // These are type-check errors / warnings by default, but cannot be.
     compilerOptions.jscomp_off.push(
       'moduleLoad', // Breaks type-only modules: google/closure-compiler#3041
-      'unknownDefines' // Closure complains about VERSION
+      'unknownDefines', // Closure complains about VERSION
+      // Build-time constant subsitution will trigger both uselessCode (DCE)
+      // as well as suspiciousCode (silly conditionals where constants are used).
```

Comment:
This shouldn't be disabled repo-wide. Functions using build-time constants are rare, they can get their own `@suppress` where the constant is used instead. Example:
https://github.com/ampproject/amphtml/blob/b8ff898def7a1597d2af019e33e27bb817b9f37a/src/core/dom/css.js#L87-L103

Reply:
I fully agree! That is a much better idea